src/execution_engine.rs

```diff
@@ -181,7 +226,13 @@ impl ExecutionEngine {
             return Err(FunctionLookupError::FunctionNotFound);
         }
 
-        Ok(address)
+        assert_eq!(size_of::<F>(), size_of::<usize>(), 
```

Comment:
This is a neat trick!

Reply:
Doesn't transmute already fail to compile if the sizes are different though?

Reply:
Yeahhh....  The `transmute_copy()` lower down makes me feel pretty dirty because it's essentially just a pointer cast. It lets us get around the *"`F` may vary in size"* transmute error, but it's kinda hacky.

Reply:
Ok; this is fine. If we find that `Symbol` conflicts with the LLVM idea of `Symbol` we can rename it when we cross that bridge I susppose.